packages/fleather/lib/src/widgets/controller.dart

```diff
@@ -17,11 +18,15 @@ List<String> _insertionToggleableStyleKeys = [
 class FleatherController extends ChangeNotifier {
   FleatherController([ParchmentDocument? document])
       : document = document ?? ParchmentDocument(),
+        history = HistoryStack.doc(document),
         _selection = const TextSelection.collapsed(offset: 0);
 
   /// Document managed by this controller.
   final ParchmentDocument document;
 
+  /// A list of changes applied to this doc. The changes could be undone or redone.
```

Comment:
Not sure we want to expose the entire `HistoryStack` object in API

Shouldn't we have a more functional API with undo/redo methods?

Reply:
I will try to move methods to controller then and hide `HistoryStack`.

Reply:
Actually, you can remove `FleatherHistory` and have an instance of `HistoryStack` in the controller (managed by the controller directly)

`Actions` and `Intent`s should be handled by the client app (we can add them to the example)

@Amir-P  WDYT?

Reply:
I think having controller to manage the stack and `Actions` in the editor talking to controller is not a bad idea. Why do you want to remove `Actions` for undo and redo from editor? @amantoux

Reply:
I had in mind that `Action`s needed to be high in the widget tree, but maybe not.
I'll try to come up with something